src/main/java/br/com/deveficiente/nossomercadolivreapi/usuario/Usuario.java

```diff
@@ -0,0 +1,43 @@
+package br.com.deveficiente.nossomercadolivreapi.usuario;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.PastOrPresent;
+import javax.validation.constraints.Size;
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@Entity
+public class Usuario {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private UUID usuarioId;
+
+    @NotEmpty(message = "Login não informado.")
+    @Email
+    private String login;
+
+    @Size(min = 6, message = "A senha deve conter no mínimo 6 caracteres")
+    private String senha;
+
+    @PastOrPresent
+    private LocalDateTime createdAt = LocalDateTime.now();
+
+    @Deprecated
+    public Usuario() {
+    }
+
+    public Usuario(@NotEmpty @Email String login, @Size(min = 6) String senha) {
```

Comment:
Ainda sobre as pistas... Como que o outro ponto do código sabe que precisa passar um password encodado aqui? Uma coisa muito complicada, principalmente quando implementamos a chamada de api e também o código do método em si é dividir a visão de fora da de dentro. Provavelmente foi óbvio para você que esse construtor recebe uma senha encodada... Mas será que seria óbvio para uma pessoa que não pareou com você? Que tipo de solução você pode usar aqui para deixar uma pista mais legal?

Será que você também pode adicionar uma defesa de borda verificando se o padrão obedece ao bcrypt?

Reply:
Verdade Alberto!
Estou muito acostumado sair fazendo como se não houvesse o amanhã hehe.

Criei a classe BCryptPassword que já faz a validação e deixa claro a intenção  no construtor, assim que for instanciar saberá que tem que mandar um objeto daquele tipo.